src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/ProjectServices/VsCoreProjectSystemReferenceReader.cs

```diff
@@ -64,26 +64,24 @@ internal class VsCoreProjectSystemReferenceReader
                 {
                     var reference3 = childReference as Reference3;
 
-                    // Set missing reference if
-                    // 1. reference is null OR
-                    // 2. reference is not resolved which means project is not loaded or assembly not found.
-                    if (reference3 == null || !reference3.Resolved)
+                    // Verify that this is a project reference
+                    if (reference3 != null && reference3.SourceProject != null)
```

Comment:
discussed offline, I believe in the past just checking reference.SourceProject when it was unresolved would result in a COM exception. For this reason we started checking .Resolved before hand